ESM TypeScript Module Resolution breaks sprinkles type definitions

### Describe the bug

We are building a component library using Vanilla Extract and Sprinkles. We have been using CommonJS module resolution via `moduleResolution: 'node'` in `tsconfig.json`. We would like to migrate to ES Module resolution via `moduleResolution: 'node16'` in `tsconfig.json`. We are encountering issues with the Sprinkles package where the types generated via ES Module resolution are not specific, when they are when using CommonJS module resolution.

When using `moduleResolution: 'node'`, our type definitions look something like this:

*dist/sprinkles.css.d.ts*
```ts
export declare const sprinkles: import("@vanilla-extract/sprinkles/dist/declarations/src/createSprinkles").SprinklesFn<[{
    conditions: {
        defaultCondition: "mobile";
        conditionNames: ("mobile" | "tablet" | "desktop")[];
    };
    styles: {
        display: {
            values: {
                block: {
                    defaultClass: string;
                    conditions: {
                        mobile: string;
                        tablet: string;
                        desktop: string;
                    };
                };
                flex: {
                    defaultClass: string;
                    conditions: {
                        mobile: string;
                        tablet: string;
                        desktop: string;
                    };
                };
                none: {
                    defaultClass: string;
                    conditions: {
                        mobile: string;
                        tablet: string;
                        desktop: string;
                    };
                };
            };
        };
    // ...
}
```

However, when we use `moduleResolution: 'node16'`, the type definition looks like this:

*dist/sprinkles.css.d.ts*
```ts
export declare const sprinkles: any;
```


### Reproduction

https://github.com/davidhousedev/vanilla-extract-esm-issue-reproduction

### System Info

```shell
System:
    OS: macOS 11.6.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 311.74 MB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Chrome: 102.0.5005.115
    Firefox: 101.0
    Safari: 15.5
  npmPackages:
    @vanilla-extract/sprinkles: ^1.4.1 => 1.4.1 
    @vanilla-extract/vite-plugin: ^3.2.1 => 3.2.1 
    vite: ^2.9.9 => 2.9.13
```


### Used Package Manager

yarn

### Logs

_No response_

### Validations

- [X] Check that there isn't [already an issue](https://github.com/seek-oss/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

I have a similar issue with `moduleResolution: "node16"`:
```
src/theme/sprinkles.css.ts(264,14): error TS2742: The inferred type of 'sprinkles' cannot be named without a reference to '../../node_modules/@vanilla-extract/sprinkles/dist/declarations/src/createSprinkles.js'. This is likely not portable. A type annotation is necessary.
```

--- a/.changeset/tasty-olives-tell.md
+++ b/.changeset/tasty-olives-tell.md
@@ -1,0 +1,5 @@
+---
+'@vanilla-extract/recipes': minor
+---
+
+Add RuntimeFn as an export to recipes index

--- a/packages/recipes/src/index.ts
+++ b/packages/recipes/src/index.ts
@@ -10,7 +10,7 @@ import type {
   VariantSelection,
 } from './types';
 
-export type { RecipeVariants } from './types';
+export type { RecipeVariants, RuntimeFn } from './types';
 
 function mapValues<Input extends Record<string, any>, OutputValue>(
   input: Input,

